lib/iris/coords.py

```diff
@@ -1762,30 +1761,42 @@ def data(self, data):
         if data is None:
             raise ValueError('The data payload of a CellMeasure may not be '
                              'None; it must be a numpy array or equivalent.')
+        if is_lazy_data(data) and data.dtype.kind in 'iu':
```

Comment:
This is still all out of scope as-of CF v1.7, but here we are covering ourselves for whatever might exist in a file, 
So yes, I'll add that as another.